### PR TITLE
Syaam/49 feeding logging

### PR DESCRIPTION
**Pull request by syaamkhandaker**
## PR Title

Issue Number(s): #49 .

What does this PR change and why?
Adds functionalities to log, check in, and feed a pet

Note: Haven't fully finished out testing of the code! I was hoping to get to this by tmrw!

### Checklist

- [ x ] Requirements have been implemented
- [ x ] Acceptance criteria is met
- [x  ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #number_of_pr

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/db/actions/pets.ts [hunk 1]
```diff
@@ -22,11 +22,33 @@ export async function getPetByUserId(userId: ObjectId) {
...
+  updateObj: { name?: string; food?: number },
+) {
   const db = client.db();
   const result = await db
     .collection("pets")
-    .updateOne({ userId: userId }, { $set: { name: name } });
+    .updateOne({ userId: userId }, { $set: { ...updateObj } });
+
+  if (result.modifiedCount == 0) {
+    throw new InternalServerError("Failed to update pet.");
```
Comment (michaelhyi) [inline src/db/actions/pets.ts:35]:
```suggestion
    throw new Error("Failed to update pet.");
```

**Review by michaelhyi — CHANGES_REQUESTED** [review]
I left some comments throughout! The biggest thing is checking for expiration, which I think is off and impacts functionality a bit.

After you fix these comments, please make sure to test the API in Postman! Thanks Syaam! great work :)

File: src/services/medication.ts [hunk 2]
```diff
@@ -164,3 +176,94 @@ export async function getMedications(userId: string) {
...
+  }
+  // check if medication check in already exists and isn't expired
+
+  const existingMedicationCheckIn = await getMedicationCheckInAction(
+    new ObjectId(medicationId),
+  );
+
+  if (existingMedicationCheckIn) {
+    const checkIn = existingMedicationCheckIn as MedicationCheckIn;
+    if (checkIn.expiration.getTime() > Date.now()) {
```
[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/services/medication.ts:201]: I fixed this issue before I saw this edit!

**Review by michaelhyi — APPROVED** [review]
